File: ios/Openid4vpBle/events/withArgs/VerificationStatusEvent.swift
import Foundation

struct VerificationStatusEvent: EventWithArgs {
    var type: String { return "onVerificationStatusReceived" }
    var status: VerificationStatus
    
    func getData() -> [String: String] {
        return ["status": status.rawValue]
    }
    
    enum VerificationStatus: Int {
        case ACCEPTED = 0
        case REJECTED = 1
    }
}

File: android/src/main/java/io/mosip/tuvali/openid4vpble/events/withArgs/VerificationStatusEvent.kt
package io.mosip.tuvali.openid4vpble.events.withArgs

import io.mosip.tuvali.openid4vpble.events.EventWithArgs

class VerificationStatusEvent(private val status: VerificationStatus): EventWithArgs {
  override fun getData(): HashMap<String, String> {
    return hashMapOf("status" to status.value.toString())
  }

  override val type = "onVerificationStatusReceived"

  enum class VerificationStatus(val value: Int) {
    ACCEPTED(0),
    REJECTED(1)
  }
}
